json-rpc/types/src/views.rs

```diff
@@ -644,18 +692,183 @@ pub struct TransactionView {
     pub vm_status: VMStatusView,
     pub gas_used: u64,
 }
+
+impl TransactionView {
+    pub fn try_from_tx_and_events(
+        version: u64,
+        tx: Transaction,
+        tx_info: TransactionInfo,
+        events: Vec<ContractEvent>,
+    ) -> Result<Self> {
+        let events = events
+            .into_iter()
+            .map(|event| EventView::try_from((version, event)))
+            .collect::<Result<Vec<_>>>()?;
+
+        Ok(TransactionView {
+            version,
+            hash: tx.hash(),
+            bytes: BytesView::new(bcs::to_bytes(&tx)?),
+            transaction: TransactionDataView::from(tx),
+            events,
+            vm_status: VMStatusView::from(tx_info.status()),
+            gas_used: tx_info.gas_used(),
+        })
+    }
+}
+
+#[derive(Clone, Debug, Deserialize, Serialize, PartialEq)]
+pub struct TransactionListView(pub Vec<TransactionView>);
+
+impl TransactionListView {
+    pub fn empty() -> Self {
+        Self(Vec::new())
+    }
+}
+
+impl TryFrom<TransactionListWithProof> for TransactionListView {
+    type Error = Error;
+
+    fn try_from(txs: TransactionListWithProof) -> Result<Self, Self::Error> {
+        if txs.is_empty() {
+            return Ok(Self::empty());
+        }
+        let start_version = txs
+            .first_transaction_version
+            .ok_or_else(|| format_err!("expected a start version since tx list non-empty"))?;
+
+        let transactions = txs.transactions;
+        let transaction_infos = txs.proof.transaction_infos;
+
+        ensure!(
+            transaction_infos.len() == transactions.len(),
+            "expected same number of transaction_infos as transactions, \
+             received {} transaction_infos and {} transactions",
+            transaction_infos.len(),
+            transactions.len(),
+        );
+
+        let events = if let Some(events) = txs.events {
```

Comment:
maybe should name this events_list, "events" is confusing as one txn has a list of events